### PR TITLE
Fix duplicated order by clause

### PR DESCRIPTION
**Pull request by moelsayed**
Currently, the `Limit` query filter is applying an `ORDER BY` clause in addition to `LIMIT`. This is causing a SQL syntax error, since `GetReconciliations()` adds another `ORDER BY` clause by default.

Unfortunately, simply removing the order clause from the filter doesn't resolve this, because the end result is also a syntactically incorrect statement. In this case the `LIMIT` clause comes _before_ the `ORDER BY` clause which is incorrect. 

Also, applying the order by clause before the filter will break other queries using different filters. For example, in cases where the query is using a `WHERE` filter, the final query would have an order by clause _before_ the where clause.

This PR implements the least invasive fix to resolve the issue.

https://github.com/kyma-project/control-plane/issues/1307

### CODE REVIEW

File: pkg/scheduler/reconciliation/persistent.go [hunk 1]
```diff
@@ -241,7 +242,11 @@ func (r *PersistentReconciliationRepository) GetReconciliations(filter Filter) (
 			return nil, errors.Wrap(err, "failed to apply sql filter")
 		}
 	}
-	recons, err := selectQ.OrderBy(map[string]string{"Created": "DESC"}).GetMany()
+	// TODO: the ORM should maintain the correct SQL statement order and handle duplicated clauses.
```
[the remark this answers is not in the document]
Comment (moelsayed) [reply pkg/scheduler/reconciliation/persistent.go:245]: https://github.com/kyma-incubator/reconciler/issues/850

**Review by pPrecel — APPROVED** [review]
LGTM

File: pkg/scheduler/reconciliation/persistent.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiscr) [reply pkg/scheduler/reconciliation/persistent.go:245]: I fear this change is causing the degreation of the processing time from 13ms to 36ms :(

**Pull request merged**